packages/bierzo-wallet/src/routes/transactions/index.tsx

```diff
@@ -7,18 +8,32 @@ import PageMenu from "../../components/PageMenu";
 import { ProcessedTx } from "../../logic/transactions/types/BwParser";
 import { BwParserFactory } from "../../logic/transactions/types/BwParserFactory";
 import { RootState } from "../../store/reducers";
+import { getChainAddressPair } from "../../utils/tokens";
 import Layout from "./components";
 import { filterTxsBy, ORDER_DESC, SortOrder, TX_DATE_COLUMN, TxsOrder } from "./components/sorting";
 
 const Transactions = (): JSX.Element => {
+  const [addresses, setAddresses] = React.useState<Address[]>([]);
   const [rows, setRows] = React.useState(5);
   const [page, setPage] = React.useState(0);
   const [orderBy, setOrderBy] = React.useState(TX_DATE_COLUMN);
   const [order, setOrder] = React.useState(ORDER_DESC);
   const parsedTxs = useSelector((state: RootState) => state.notifications.transactions);
+  const identities = useSelector((state: RootState) => state.extension.identities);
+
+  React.useEffect(() => {
+    async function processIdentities(identities: { [chain: string]: Identity }): Promise<void> {
+      setAddresses((await getChainAddressPair(identities)).map(pair => pair.address));
```

Comment:
Oh, this function should be called `getChainAddressPairs` (plural) since it returns many pairs